src/image/loading_displaying.js

```diff
@@ -461,7 +542,43 @@ p5.prototype.image = function(
   _sh *= pd;
   _sw *= pd;
 
-  const vals = canvas.modeAdjust(_dx, _dy, _dw, _dh, this._renderer._imageMode);
+  let vals = { x: _dx, y: _dy, h: _dh, w: _dw };
+  if (fit) {
+    if (fit === constants.COVER) {
+      const { x, y, w, h } = _imageCover(
+        img,
+        xAlign,
+        yAlign,
+        _dw,
+        _dh,
+        _sx,
+        _sy,
+        _sw,
+        _sh
+      );
+      _sx = x;
+      _sy = y;
+      _sw = w;
+      _sh = h;
+    } else if (fit === constants.CONTAIN) {
+      const { x, y, w, h } = _imageContain(
+        xAlign,
+        yAlign,
+        _dx,
+        _dy,
+        _dw,
+        _dh,
+        _sw,
+        _sh
+      );
+      vals.x = x;
+      vals.y = y;
+      vals.w = w;
+      vals.h = h;
+    }
+  } else {
```

Comment:
`imageMode` shouldn't affect the image shown when `fit` mode is specified.

Reply:
I think we still want `imageMode` to affect positioning when `fit` is specified, and it should just refine the positioning. of the source image within the destination box specified. e.g. starting from this non-`fit` code:

<table>
<tr>
<td>

```js
rectMode(CENTER)
imageMode(CENTER)
rect(0, 0, dw, dh)
image(img, 0, 0, dw, dh, 0, 0, sw, sh)
```

</td>
<td>
<img src="https://user-images.githubusercontent.com/5315059/188283217-21bab2cc-1a3f-4f9a-8ef1-06062d3ec32c.png">
</td>
</tr>
</table>

adding some fit parameters, I would expect it to look like this:

<table>
<tr>
<td>

```js
rectMode(CENTER)
imageMode(CENTER)
rect(0, 0, dw, dh)
image(img, 0, 0, dw, dh, 0, 0, sw, sh, COVER, RIGHT, CENTER)
```

</td>
<td>
<img src="https://user-images.githubusercontent.com/5315059/188283306-e1b48831-6e68-4c73-99ac-c9ca22a52331.png">
</td>
</tr>
</table>

but without taking into account `imageMode`, it looks like this currently:

<table>
<tr>
<td>

```js
rectMode(CENTER)
imageMode(CENTER)
rect(0, 0, dw, dh)
image(img, 0, 0, dw, dh, 0, 0, sw, sh, COVER, RIGHT, CENTER)
```

</td>
<td>
<img src="https://user-images.githubusercontent.com/5315059/188283261-ea161c5d-b524-4f49-ae68-7abfa8441840.png">
</td>
</tr>
</table>

Reply:
Currently, it doesn't take `imageMode` into account and produces the result as if `imageMode` is always `CORNER`. I was not able to reproduce this image going out of bound. I would appreciate if you could send me the `src` for this sketch. Also, my question regarding `imageMode` is if `imageMode` is `CENTER` and the user inputs 
`image(img, 0, 0, dw, dh, 0, 0, sw, sh, COVER, RIGHT, CENTER` should we consider `yAlign: center` from the center of the image.